tests/integration/test_settings.py

```diff
@@ -66,15 +72,25 @@ def test_setting_origin_code(test_agent_session, run_python_code_in_subprocess):
             "DD_LOGS_INJECTION": "true",
             "DD_TRACE_HEADER_TAGS": "X-Header-Tag-1:header_tag_1,X-Header-Tag-2:header_tag_2",
             "DD_TAGS": "team:apm,component:web",
+            "DD_TRACE_ENABLED": "true",
         }
     )
     out, err, status, _ = run_python_code_in_subprocess(
         """
 from ddtrace import config, tracer
+
+# unsubscribe listeners to the "_tracing_enabled" config change event
+# because the tracer uses such a listener to disable itself
+# the tracer controls the sending of telemetry events, so when it's disabled none of these
+# events are sent
```

Comment:
It is a requirement of the RFC to have the telemetry events set. Do you think it's possible to do so in this PR? I think this is a good iteration on the feature so I'm good with merging this and following up for the telemetry work (so long as we can ensure both are released at the same time).

Reply:
I pulled out the functional bits from this pull request to https://github.com/DataDog/dd-trace-py/pull/8216, which adds new code paths without providing a way to execute them (the capability is not reported to the RC backend). That change is mergeable, and I'll keep this PR #8080 open as I fix telemetry.